src/tests/test_split.cpp

```diff
@@ -0,0 +1,150 @@
+                                                               dif_table_has_header);
+        return {{kCsvConfig, csv_config}, {kDifferenceTable, std::move(table)}};
+    }
+
+    static std::unique_ptr<algos::dd::Split> CreateSplitAlgorithmInstance(
+            CSVConfig const& csv_config, std::filesystem::path const& dif_table_path = "",
+            char dif_table_separator = ',', bool dif_table_has_header = true) {
+        return algos::CreateAndLoadAlgorithm<algos::dd::Split>(
+                GetParamMap(csv_config, dif_table_path, dif_table_separator, dif_table_has_header));
+    }
+};
+
+TEST_F(SplitAlgorithmTest, Test0) {
+    auto algo = CreateSplitAlgorithmInstance(
+            kTestDD, std::filesystem::current_path() /
+                             "../../src/core/algorithms/dd/split/datasets/TestDif.csv");
+    algo->Execute();
+
+    auto actual_results = algo->GetDDStringList();
+    std::set<std::pair<std::set<model::DFStringConstraint>, std::set<model::DFStringConstraint>>>
+            expected_results = {{{{"Col4", 2, 4}}, {{"Col0", 3, 4}}},
+                                {{{"Col1", 2, 5}}, {{"Col0", 1, 1}}}};
+    CompareDDStringLists(expected_results, actual_results);
+}
+
+TEST_F(SplitAlgorithmTest, Test1) {
+    auto algo = CreateSplitAlgorithmInstance(kTestDD1);
+    algo->Execute();
+
+    auto actual_results = algo->GetDDStringList();
+    std::set<std::pair<std::set<model::DFStringConstraint>, std::set<model::DFStringConstraint>>>
+            expected_results = {{{{"Col1", 2, 3}}, {{"Col0", 1, 1}}},
+                                {{{"Col0", 1, 1}}, {{"Col1", 2, 2}}}};
+    CompareDDStringLists(expected_results, actual_results);
+}
+
+TEST_F(SplitAlgorithmTest, Test2) {
+    auto algo = CreateSplitAlgorithmInstance(
+            kTestDD2, std::filesystem::current_path() /
+                              "../../src/core/algorithms/dd/split/datasets/TestDif1.csv");
+    algo->Execute();
+
+    auto actual_results = algo->GetDDStringList();
+    std::set<std::pair<std::set<model::DFStringConstraint>, std::set<model::DFStringConstraint>>>
+            expected_results = {{{{"Col3", 5, 5}}, {{"Col2", 4, 4}}}};
+    CompareDDStringLists(expected_results, actual_results);
+}
+
+TEST_F(SplitAlgorithmTest, Test3) {
+    auto algo = CreateSplitAlgorithmInstance(
+            kTestDD2, std::filesystem::current_path() /
+                              "../../src/core/algorithms/dd/split/datasets/TestDif2.csv");
+    algo->Execute();
+
+    auto actual_results = algo->GetDDStringList();
+    std::set<std::pair<std::set<model::DFStringConstraint>, std::set<model::DFStringConstraint>>>
+            expected_results = {{{{"Col3", 7, 12}}, {{"Col1", 1, 1}}},
+                                {{{"Col3", 5, 5}}, {{"Col1", 2, 2}}},
+                                {{{"Col3", 5, 7}, {"Col2", 4, 4}}, {{"Col1", 2, 2}}},
+                                //{{{"Col3", 5, 5}}, {{"Col2", 4, 4}}},
```

Comment:
Why some  lines are commented?

Reply:
It turned out that there are two possible minimal covers for some tests because there are two options of safely deleting a transitively implied DD. Comments are added for easier understanding of which DD is deleted and which is present.

Reply:
Could we leave some explanation in comments about it?